### DISCUSSION TITLE
How to change color with custom icon?

### DISCUSSION THREAD

**Discussion by EpsilonYi** [vuetifyjs/vuetify#18750] [Q&A]
I customize a icon component 
`
    function filename(path: string) {
      return path
        .split(/(\\|\/)/g)
        .pop()!
        .replace(/\.[^/.]+$/, '')
    }
    
    const svgIcons = Object.fromEntries(
      Object.entries(
        import.meta.glob('@/assets/icons/*.svg', {
          eager: true,
          as: 'raw',
        }),
      ).map(([k, v]) => [filename(k), v]),
    )
    
    const custom: IconSet = {
      component: (props: IconProps) => {
         return h(props.tag, {
           innerHTML: svgIcons[props.icon as string],
        })
      }
    }
    
    export {custom}
`

when i use it lick this <VIcon icon={'custom:logo'} color="#aabbee" /> , the color

**Comment by nickmadev** [comment]
It seems like you are trying to apply random colors with your input file.
In my opinion, it is impossible in vuetify or vue.js level.
You need additional codes in the level of image analysis inside your code.

**Comment by nickmadev** [comment]
@EpsilonYi 
Please try this additional code.
```
     return h(props.tag, {
       innerHTML: svgIcons[props.icon as string],
       style: {
         color: props.color,
       },
    })
```

**Reply by EpsilonYi** [reply]
Thanks for answer，but it not work.I try to add a class name 'v-icon__svg' to inner element, it can contrl color. but how to use h function to do this?
